src/WinRT.Runtime/FundamentalMarshalers.cs

```diff
@@ -67,5 +71,183 @@ internal struct Char
         public static void DisposeMarshaler(char m) { }
         public static void DisposeAbi(ushort abi) { }
     }
+
+    // This partial declaration contains additional marshalling methods for the ABI type for System.Type.
+    // These are hardcoded for the ABI.System.Type type, which makes them AOT-safe. Previously, these were
+    // relying on AOT-unfriendly code in MarshalNonBlittable<T>. Because this is a well known type, we can
+    // just hardcode explicit marshalling methods for it instead to make using this type fully AOT-safe.
+    partial struct Type
```

Comment:
Any advantages to putting this here rather than in `Type.cs` where `CreateMarshaler` and so on are already there.

Reply:
No specific reason other than I wanted to separate the definition from the rest of the stuff on the ABI type, as these methods are just internal helpers and they're not public. Maybe we could just add a `Type.ArrayMarshalling.cs` partial file? Otherwise I can also just move these to the same file if you have a strong preference 🙂

Reply:
`Type.ArrayMarshalling.cs` might be better then.  It just sometimes it gets hard to find where certain marshalers are defined and it can get confusing when you see some defined in one file and someone might think that is all of them.

Reply:
Yup, no that makes perfect sense! Done in 9f265c3c31e45bb981887cde877fd32c71fc1223 🙂